### PR TITLE
Use brigade sdk instead of k8s api 

### PR DESCRIPTION
**Pull request by carolynvs**
* Execute jobs by requesting them from the Brigade API instead of creating pods directly.
* Adds circleci until we can use brigade2 with drake for our own build

### CODE REVIEW

**Comment by carolynvs** [comment]
@krancour I'd like to test this out locally before merging. Which is the right script to build the brigdrake worker docker image?

**Comment by krancour** [comment]
You have devdrake installed? Try `drake list` to find the jobs. There should be jobs for building the images using "dood" or "dind". Locally, "dood" is a better choice.

**Comment by krancour** [comment]
Best PR ever.

Screen Shot 2021-04-09 at 7 48 21 PM

File: brigdrake-values.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+################################################################################
+## The logic for conditionally including subcharts of other subcharts seems not
+## to work as one might expect. As a workaround, we move sub-sub-chart
+## enable/disable flags up a level from where we might otherwise expect to
+## find them.
+################################################################################
+kasthi:
+  enabled: true
+brigade-github-app:
+  enabled: true
```
Comment (krancour) [inline brigdrake-values.yaml:160]: This file looks like it was included accidentally.

File: byu-kube.config [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        tenant-id: cc18ecf3-7acb-4d14-ba43-fc25dc310191
+      name: azure
```
Comment (krancour) [inline byu-kube.config:29]: This appears to be another accidental inclusion.

File: pkg/brigade/drakespec/convert.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package drakespec
+
+import (
+	"fmt"
+
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/lovethedrake/drakecore/config"
+)
+
+func ToBrigadeJob(pipelineName string, eventID string, jobDef config.Job) core.Job {
```
Comment (krancour) [inline pkg/brigade/drakespec/convert.go:10]: I like this idea of just "translating." Drake job in, Brigade job out. 👍

File: pkg/brigade/event.go [hunk 4]
```diff
@@ -33,31 +74,16 @@ type Revision struct {
...
-	}
-}
-
-// GetEventFromEnvironment returns an Event object with values derived from
-// environment variables.
-func GetEventFromEnvironment() (Event, error) {
-	e := NewEventWithDefaults()
-	err := envconfig.Process("", &e)
+// GetEventPayload returns an EventPayload object with values derived from
+// /var/event/event.json
```
Comment (krancour) [inline pkg/brigade/event.go:78]: Just be careful with the term "payload" because it might be overloaded. Events have a _field_ called payload (which is presumably the original payload from whatever was received by the gateway from some upstream source). It could be confusing to also call an Event itself a "payload." _I_ get what you mean, but can't guarantee others won't be confused by the overload.

File: pkg/brigade/executor/pipeline_executor.go [hunk 5]
```diff
@@ -28,6 +23,7 @@ func executePipeline(
 
 	// If ANY of the pipeline's jobs' containers mounts shared storage, we need to
 	// create a volume.
+	// TODO(carolynvs): Determine if we need this check or if the substrate handles it now
```
Comment (krancour) [inline pkg/brigade/executor/pipeline_executor.go:26]: You don't need this check.

This is a major difference between Brigade 1.x and Brigade 2. In 1.x, the worker image was responsible for provisioning volumes like the shared workspace. A terrible side effect of that was that that volume wasn't mounted to the worker itself. Jobs could all read from and write to the shared workspace, but the worker couldn't-- which was, at times, inconvenient.

In Brigade 2, a project definition will say whether a shared workspace should be provisioned or not. If so, that happens _before_ the worker is launched. (Which means it can be, and is, mounted to the worker itself.)

Long story short, if a shared volume needs to be provisioned, it's already happened by this point.

File: pkg/brigade/project.go [hunk 6]
```diff
@@ -1,113 +1,5 @@
...
 
 import (
-	"context"
-	"encoding/json"
-	"strings"
-
-	"github.com/kelseyhightower/envconfig"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
+	_ "github.com/brigadecore/brigade/sdk/v2"
```
Comment (krancour) [inline pkg/brigade/project.go:4]: What is this doing? There shouldn't be any side effects I know of from loading that package.

File: pkg/drake/brig/trigger.go [hunk 7]
```diff
@@ -22,10 +22,10 @@ func NewTriggerFromJSON(jsonBytes []byte) (drake.Trigger, error) {
 }
 
 func (t *trigger) Matches(event brigade.Event) (bool, error) {
-	if event.Provider != "brigade-cli" {
+	if event.Source != "brigade-cli" {
```
Comment (krancour) [inline pkg/drake/brig/trigger.go:25]: In Brigade 2, this source for events created from the CLI is `brigade.sh/cli`

File: pkg/drake/github/pull_request_event_selector.go [hunk 8]
```diff
@@ -23,7 +23,7 @@ func (p *pullRequestEventSelector) matches(
 		return false, nil
 	}
 	pre := github.PullRequestEvent{}
-	if err := json.Unmarshal(event.Payload, &pre); err != nil {
+	if err := json.Unmarshal([]byte(event.Payload), &pre); err != nil {
```
Comment (krancour) [inline pkg/drake/github/pull_request_event_selector.go:26]: Purely fyi, I don't like how I did the github triggers. There were things about github's APIs that I understand now that I didn't understand then. I definitely want to revisit this... and then extract a tigger spec from it.

**Comment by krancour** [comment]
I am so excited for this PR!

File: pkg/brigade/project.go (hunk 6 above)
Comment (carolynvs) [reply pkg/brigade/project.go:4]: oops! This was the first project I edited and was trying to force go mod to download the package so that I could get autocomplete in the editor. Just forgot to take that line out.

This is totally just a WIP and isn't complete yet. Sorry for not reviewing it more closely before pushing!

**Comment by krancour** [comment]
🎉  🎉  🎉  🎉  🎉
